src/signal_manager.sh

```diff
@@ -0,0 +1,48 @@
+. "$KW_LIB_DIR/kwio.sh" --source-only
+
+declare -ga SIGNAL_MANAGER_VALID_SIGNALS=( $(trap -l | sed -E 's/[0-9]+\)/ /g') )
+
+# This functions prints the default message when interrupting kw
+function default_interrupt_handler()
+{
+  say $'\n'Oh no! An interruption! See ya...
+}
+
+# This function adds a new signal handler to an arbitrary signal
+#
+# @command The handler to be called when a signal is catched
+# @signals The signals to attach to this handler
+#
+# If command is empty, use the default handler
+# If the signal is empty, use the default signals
+
+function signal_manager()
+{
+  local command="$1"
+  shift
+  local -a signals=( "$@" )
+
+  if [[ -z "$command" ]]; then
+    command='default_interrupt_handler'
+  fi
+
+  if [[ "${#signals}" = 0 ]]; then
+    signals=( SIGINT SIGTERM )
+  else
+    for s in "${signals[@]}"; do
+      if [[ ! " ${SIGNAL_MANAGER_VALID_SIGNALS[@]} " =~ " $s " ]]; then
+        return 22 # EINVAL
+      fi
+    done
+  fi
+
+  trap "$command" "${signals[@]}" 2>/dev/null
+}
+
+# This function resets all signal handlers to their system defaults and
+# sets kw's default signal handler for SIGINT and SIGTERM
+function signal_manager_reset()
+{
+  trap | grep -o '[^ ]*$' | while read sig; do trap - $sig; done
```

Comment:
That's a very clever line, but it is a little bit hard to understand... maybe you can expand it in small steps to make it easy to read.